salt/transport/zeromq.py

```diff
@@ -577,19 +577,20 @@ def _init_socket(self):
         self.stream = zmq.eventloop.zmqstream.ZMQStream(
             self.socket, io_loop=self.io_loop
         )
+        self.stream.on_recv(self.handle_reply)
 
-    def timeout_message(self, message):
+    def timeout_message(self, future):
         """
         Handle a message timeout by removing it from the sending queue
         and informing the caller
 
         :raises: SaltReqTimeoutError
         """
-        future = self.send_future_map.pop(message, None)
-        # In a race condition the message might have been sent by the time
-        # we're timing it out. Make sure the future is not None
-        if future is not None:
-            future.set_exception(SaltReqTimeoutError("Message timed out"))
+        try:
+            self._future = None
```

Comment:
Why would setting an attribute to `None` raise a `ValueError`?

Reply:
fixed